### DISCUSSION TITLE
Streamlink -O audio_only  => FFMPEG -> mp3 requires babysitting...?

### DISCUSSION THREAD

**Discussion by RickKukiela** [streamlink/streamlink#3608] [Q&A]
I'm hoping someone can offer some insight as to what I'm doing wrong here. I'm a programmer and I work all day at my computer and I like listening to music all day. There is a LOT of hours so when I found out a bunch of my favorite djs' are playing hours of live sets every week on twitch I was delighted. However I quickly was bummed because during the day when they are not "live" the playbacks are riddled with muted sections because of copyright bs. 

So I found streamlink and set up a task scheduler to run a bat file I created which basically takes some arguments and then runs a streamlink command to copy the stream directly to an MP3 file which is saved in a location that is automatically included into my plex library for easy playback during the day.

My bat file runs this command:

```
streamlink --twitch-disable-ads -O %3 audio_only | ffmpeg -i pipe:0 -c:v none -c:a libmp3lame -b:a 320k -joint_stereo 0 -y -format mp3 -metadata artist="%artist%" -metadata title="%title%" -metadata album="%show%" D:\sync\stream-rips\%fn%
```
So I can record a show using the collowing command:
```
rip-stream.bat <streamUrlId> <baseFileNameForMp3> <twitchUrlToBroadcast> <ArtistName> <ShowName>
```

This all works really well except for on really weird issue...

The cmd window where the record is happening needs to be "babysat" or it will just sort of stop writing the buffer to the file. What I mean is that it will look something like this after a bit:

![Screenshot_2](https://user-images.githubusercontent.com/3813238/110174718-256f9b80-7dc6-11eb-9222-28472f78307b.jpg)

If you notice the bottom right corner it shows the speed at around 1x which is what you want... but what's weird is that the "time=" portion will say a less amount of time than the record has been running. Say its been running for an hour, I could come back and it will say its only been running for 30 minutes, but speed is 1x.... so if i click the window and press the enter key, it wakes up all of the sudden and starts writing the full buffer to the file, the 1x will immutably jump down to, in this example .5x which is 50% speed, ie 30 minutes of 1 hour. But the buffer has maybe 10 minutes of audio so the 1x drops to .5x then quickly goes up to like .65x or whatever the time in the buffer makes up...

So when that happens, at that point, it keeps working normally for a while, but the resulting recording will miss and jump cut from like 40 minutes (30 minutes + 10 minute buffer) to what was being played at the 60 minute mark (when I pressed enter).

I'm not sure what causes this to happen. It might be a commercial happening mid stream or something. It usually either happens for the first 10 minutes of a set i'm recording, or it will just happen randomly after a few hours.

Generally I also have twitch open in my browser during the recordings.

Anyone know whats going on here?

**Comment by h1z1** [comment]
Streamlink is not able to feed ffmpeg, likely due to either source problems which you'd need to look at the original to determine, or hardware.

Since you're for some reason re-encoding aac to mp3, the time is the output from ffmpeg not streamlink.

**Comment by beardypig** [comment]
Have you tried saving the stream to a file and then transcoding it later (although you could probably just let plex do that if required). I believe audio_only from twitch will be in an m4a container.
